omnigibson/utils/registry_utils.py

```diff
@@ -351,28 +434,47 @@ def _load_state(self, state):
         # the state might contain additional information about objects that are NOT in the scene. For both cases, state
         # loading will be skipped.
         for obj in self.objects:
-            if obj.name not in state:
-                log.warning(f"Object '{obj.name}' is not in the state dict to load from. Skip loading its state.")
-                continue
-            obj.load_state(state[obj.name], serialized=False)
+            if self._load_filter(obj):
+                if obj.name not in state:
+                    log.warning(f"Object '{obj.name}' is not in the state dict to load from. Skip loading its state.")
+                    continue
+                obj.load_state(state[obj.name], serialized=False)
 
     def serialize(self, state):
         # Iterate over the entire dict and flatten
-        return (
-            np.concatenate([obj.serialize(state[obj.name]) for obj in self.objects])
-            if len(self.objects) > 0
+        # We keep track of how many objects are being saved, as well as the unique identifier for each object so that
+        # the saved flattened array is agnostic to object ordering
+        n_objs = len(state)
+        state_flat = (np.concatenate([np.insert(self("name", name).serialize(state[name]), 0, getattr(self("name", name), self.hash_key)) for name in state])
+            if len(state) > 0
```

Comment:
hmm not sure I fully understand, maybe we can discuss?

Reply:
can we make this more readable.

Reply:
updated